pkg/commands/handler.go

```diff
@@ -0,0 +1,246 @@
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/fly-examples/postgres-ha/pkg/flypg"
+	"github.com/fly-examples/postgres-ha/pkg/flypg/admin"
+	"github.com/fly-examples/postgres-ha/pkg/render"
+	"github.com/go-chi/chi/v5"
+	"github.com/jackc/pgx/v4"
+)
+
+func Handler() http.Handler {
+	r := chi.NewRouter()
+
+	r.Route("/users", func(r chi.Router) {
+		r.Get("/{name}", handleFindUser)
+		r.Get("/list", handleListUsers)
+		r.Post("/create", handleCreateUser)
+		r.Delete("/delete/{name}", handleDeleteUser)
+	})
+
+	r.Route("/databases", func(r chi.Router) {
+		r.Get("/list", handleListDatabases)
+		r.Get("/{name}", handleFindDatabase)
+		r.Post("/create", handleCreateDatabase)
+		r.Delete("/delete/{name}", handleDeleteDatabase)
+	})
+
+	return r
+}
+
+func handleListDatabases(w http.ResponseWriter, r *http.Request) {
+	pg, close, err := getConnection(r.Context())
+	if err != nil {
+		render.Err(w, err)
+		return
+	}
+	defer close()
+
+	dbs, err := admin.ListDatabases(r.Context(), pg)
+	if err != nil {
+		render.Err(w, err)
+		return
+	}
+	res := &Response{
+		Result: dbs,
+	}
+
+	render.JSON(w, res, http.StatusOK)
+}
+
+func handleListUsers(w http.ResponseWriter, r *http.Request) {
+	pg, close, err := getConnection(r.Context())
+	if err != nil {
+		render.Err(w, err)
+		return
+	}
+	defer close()
+
+	users, err := admin.ListUsers(r.Context(), pg)
+	if err != nil {
+		render.Err(w, err)
+		return
+	}
+	res := &Response{
+		Result: users,
+	}
+
+	render.JSON(w, res, http.StatusOK)
+
+}
+
+func handleFindUser(w http.ResponseWriter, r *http.Request) {
+	pg, close, err := getConnection(r.Context())
+	if err != nil {
+		render.Err(w, err)
+		return
+	}
+	defer close()
+
+	name := chi.URLParam(r, "name")
+
+	user, err := admin.FindUser(r.Context(), pg, name)
+	if err != nil {
+		render.Err(w, err)
+		return
+	}
+	res := &Response{
+		Result: user,
+	}
+	render.JSON(w, res, http.StatusOK)
+}
+
+func handleFindDatabase(w http.ResponseWriter, r *http.Request) {
+	pg, close, err := getConnection(r.Context())
+	if err != nil {
+		render.Err(w, err)
+		return
+	}
+	defer close()
+
+	name := chi.URLParam(r, "name")
+
+	db, err := admin.FindDatabase(r.Context(), pg, name)
```

Comment:
This seems to return status code 500 when the the target database doesn't exist.

Flyctl expects 404: https://github.com/superfly/flyctl/blob/be327db0075b8e4bb35eda4c8359da0936b0c42d/pkg/flypg/pg.go#L92